BGAnimations/ScreenGameplay underlay/PerPlayer/NoteField/DisplayMods.lua

```diff
@@ -1,16 +1,23 @@
 local player = ...
 
-if SL.Global.GameMode == "Casual" or GAMESTATE:IsCourseMode() then return end
+if SL.Global.GameMode == "Casual" then return end
 
 local optionslist = GetPlayerOptionsString(player)
 
 local af = Def.ActorFrame{
   InitCommand = function(self)
-      self:xy(GetNotefieldX(player), SCREEN_HEIGHT/4*1.3)
+    self:diffusealpha(1):xy(GetNotefieldX(player), SCREEN_HEIGHT/4*1.3)
   end,
   OnCommand=function(self)
     self:sleep(5):decelerate(0.5):diffusealpha(0)
-  end
+  end,
+  CurrentSongChangedMessageCommand=function(self)
+    self:stoptweening():playcommand("Init"):queuecommand("On")
```

Comment:
Behavior question: Do you think the player should be displayed their mods on every song in course mode? Or only on the first one and then again *if* they're changing mods?

I can potentially see how displaying mods on every song would often be redundant and I'm unsure if it would be bothersome for some charts that start really fast.

Reply:
Yeah, now that you point it out, I agree that displaying mods at the start of *every* CourseMode song is overkill.  I'll update this PR to only show mods
1. at start of 1st song
2. subsequently, if SpeedMod is changed

when I have a moment.

Reply:
All right, thanks for your patience on this! :)

I force-pushed 1 update to remove the use of `CurrentSongChangedMessageCommand` and a 2nd update to condense two `string.format` operations into one.